python/ray/data/read_api.py

```diff
@@ -394,6 +443,7 @@ def read_datasource(
         inmemory_size,
         block_list._estimated_num_blocks,
         ray_remote_args,
+        concurrency,
```

Comment:
Sorry, there's a bug here I miss to pass the `concurrency` parameter in previous PR. The added unit test should catch it in the future. cc @raulchen.